komodo/matrix.py

```diff
@@ -6,35 +6,53 @@
 
 import itertools
 import re
-from typing import Iterator, Sequence, Tuple
+from typing import Dict, Iterator, Optional, Sequence, Tuple
 
 
 def get_matrix(
     rhel_versions: Sequence[str],
     py_versions: Sequence[str],
-) -> Iterator[Tuple[str, str]]:
+    other_components: Optional[Dict[str, Sequence[str]]] = None,
```

Comment:
other_components -> other_component unless we plan to support more than three dimensions?

Reply:
I should limit this to just one extra level. Anything else is ludicrous.